server/engine/src/main/java/org/eclipse/lsp/cobol/service/DocumentModelService.java

```diff
@@ -178,17 +180,37 @@ public boolean isDocumentSynced(String uri) {
   @Synchronized
   public Map<String, List<Diagnostic>> getOpenedDiagnostic() {
     Map<String, List<Diagnostic>> result = new HashMap<>();
+
     for (Map.Entry<String, CobolDocumentModel> entry : docs.entrySet()) {
-      String uri = entry.getKey();
+      String path = getPathFromUri(entry.getKey());
+      if (path == null) continue;
+
       CobolDocumentModel document = entry.getValue();
-      List<Diagnostic> diagnostics = diagnosticRepo.get(document.getUri());
-      if (diagnostics != null && document.isOpened()) {
-        result.put(uri, diagnostics);
-      } else {
-        result.put(uri, ImmutableList.of());
-      }
+      List<Diagnostic> diagnostics = getDiagnosticsForPath(path);
+
+      result.put(entry.getKey(), document.isOpened() ? diagnostics : Collections.emptyList());
     }
 
     return result;
   }
+
+  private String getPathFromUri(String uriString) {
+    try {
+      return new URI(uriString).getPath();
```

Comment:
I am not sure, if this is the best way to compare uri's. But seems this works

Reply:
This ignores both the scheme and the authority, doesn't it?